supabase/functions/updates/index.ts

```diff
@@ -34,6 +61,25 @@ async function main(url: URL, headers: BaseHeaders, method: string, body: AppInf
     return update(body)
   }
 
+  const ipAddress = [headers['x-real-ip'], headers['x-forwarded-for'], headers['client-ip']].find(i => i !== undefined && i !== null)
+  if (ipAddress) {
+    const getCurrentCount = await getDeviceRequestCount(ipAddress)
+    if (getCurrentCount) {
+      if (getCurrentCount === -1) {
+        return sendRes({ error: `Blacklist` }, 403)
+      }
+      // TODO: Update limit to blacklist flag
+      else if (getCurrentCount > 10 ** 5) {
+        blacklistDeviceRequest(ipAddress)
+        return sendRes({ error: `Blacklist` }, 403)
+      }
+      await incrementDeviceRequestCount(ipAddress);
+    }
+    else {
+      await incrementDeviceRequestCount(ipAddress);
+    }
```

Comment:
If on the else we'll run the `incrementDeviceRequestCount` anyways, we could do it outside the condition block instead

```suggestion
    }
    await incrementDeviceRequestCount(ipAddress);
```